dispatch/static/manager/src/js/vendor/dispatch-editor/embeds/TwitterEmbed.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+
+import fetchJsonp from 'fetch-jsonp'
+import url from 'url'
+
+import { Button } from '@blueprintjs/core'
+import { FormInput, TextInput } from '../../../components/inputs'
+
+require('isomorphic-fetch')
+require('../styles/embeds/video.scss')
+
+const TWITTER_API_URL = 'https://api.twitter.com/1.1/statuses/oembed.json'
+
+class TwitterEmbedComponent extends React.Component {
+
+  validateTweetURL(url) {
+    return /^(?:https?:\/\/)?(?:www\.)?twitter\.com\/?(\w+)\/(status)\/(\d+)/g.test(url)
+  }
+
+  getTweet() {
+    const that = this
+    if (this.validateTweetURL(this.props.data.url)) {
+      fetchJsonp(TWITTER_API_URL + url.format({query: {url: this.props.data.url}}))
+      .then((response) => response.json())
+      .then(data => {
+        that.props.updateField('tweet', data.html)
+        this.props.stopEditing()
+      })
+    }
+  }
+
+  renderInput() {
+    return (
+      <div className='o-embed o-embed--tweet'>
+        <form>
+          <FormInput label='URL'>
+            <TextInput
+              fill={true}
+              value={this.props.data.url}
+              onChange={e => this.props.updateField('url', e.target.value)} />
+          </FormInput>
+        </form>
+        <div className='o-embed--video__button__container'>
+          <Button onClick={() => this.getTweet()}>Insert</Button>
+        </div>
+      </div>
+    )
+  }
+
+  renderTweet() {
+    return (
+      <div>
```

Comment:
I don't think you need to have two divs here.

Also, can you add `1 rem` of padding around this div so that the tweet is better formatted?

Reply:
We might need to see this on Wednesday as those div are actually not shown on the DOM, I have taken care of the other requests in the meantime